microsetta_admin/server.py

```diff
@@ -556,6 +623,125 @@ def new_kits():
+                     as_attachment=True,
+                     download_name=fname)
+
+
+@app.route('/add_barcode_to_kit', methods=['GET', 'POST'])
+def new_barcode_kit():
+    if request.method == 'GET':
+        return render_template('add_barcode_to_kit.html',
+                               **build_login_variables())
+
+    kit_ids = []
+    barcodes = []
+
+    if 'kit_ids' in request.form:
+        kit_ids = request.form['kit_ids']
+
+    if 'user_barcode' in request.form:
+        user_barcode = request.form['user_barcode']
+        barcodes.append(user_barcode)
+    else:
+        user_barcode = None
+
+    generate_barcode_single = \
+        request.form.get('generate_barcode_single')
+    generate_barcodes_multiple = \
+        request.form.get('generate_barcodes_multiple')
+
+    if 'kit_ids' in request.files:
+        kit_ids_file = request.files['kit_ids']
+        kit_ids = _read_csv_file(kit_ids_file)
+
+    if 'barcodes_file' in request.files:
+        barcodes_file = request.files['barcodes_file']
+        barcodes = _read_csv_file(barcodes_file)
+
+    if generate_barcode_single or user_barcode:
+        if not user_barcode:
+            payload = {
+                'action': 'create',
+                'kit_ids': kit_ids,
+                'generate_barcode_single': generate_barcode_single
+            }
+            barcodes = _handle_add_barcodes_api_request(payload)
+            if isinstance(barcodes, tuple):
+                return barcodes
+
+        payload = {
+            "action": "insert",
+            "barcodes": barcodes,
+            "kit_ids": kit_ids
+        }
+        result = _handle_add_barcodes_api_request(payload)
+        if isinstance(result, tuple):
+            return result
+
+        return render_template('add_barcode_to_kit.html',
+                               barcodes=barcodes,
+                               kit_id=kit_ids,
+                               **build_login_variables())
+
+    if generate_barcodes_multiple:
+        payload = {
+            'action': 'create',
+            'kit_ids': kit_ids,
+            'generate_barcodes_multiple': generate_barcodes_multiple
+        }
+        barcodes = _handle_add_barcodes_api_request(payload)
+        if isinstance(barcodes, tuple):
+            return barcodes
+
+    if len(kit_ids) != len(barcodes):
+        error_message = f'The number of kit IDs ({len(kit_ids)}) '\
+                        f'does not match the number of barcodes '\
+                        f'({len(barcodes)}).'
+        return render_template('add_barcode_to_kit.html',
+                               error_message=error_message,
+                               **build_login_variables()), 400
+
+    payload = {
+        "action": "insert",
+        "barcodes": barcodes,
+        "kit_ids": kit_ids
+    }
+    result = _handle_add_barcodes_api_request(payload)
+    if isinstance(result, tuple):
+        return result
+
+    return _save_and_send_csv(kit_ids, barcodes)
```

Comment:
This code would be considerably more readable if it was an if/elif block, rather than cascading through conditional return statements that require cross-referencing which variable is being used.